### PR TITLE
IGAP-628: Add jpal translations and adjustments

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: native/src/routes/JpalTracking.tsx [hunk 1]
```diff
@@ -30,43 +32,94 @@ const DescriptionContainer = styled.TouchableOpacity`
...
+          return
+        }
+        if (trackingCode?.length === TRACKING_CODE_LENGTH && trackingEnabled) {
+          // Tracking enabled and (hopefully) valid code entered, so we are good to leave the screen
+          return
+        }
+
+        e.preventDefault()
+
+        Alert.alert(t('trackingLeaveTitle'), t('trackingLeaveDescription', { appName: buildConfig().appName }), [
```
Comment (klinzo) [inline native/src/routes/JpalTracking.tsx:96]: Not beautiful to use an alert here but we need some buttons so the snackbar would not work.

File: translations/translations.json [hunk 2]
```diff
@@ -5033,9 +5033,15 @@
       "tracking": "J-PAL Evaluation",
       "allowTracking": "Tracking zulassen",
       "trackingCode": "Tracking Code",
-      "trackingDescription": "Durch das Senden von Tracking-Daten Integreat langfristig verbessern.",
+      "trackingShortDescription": "Nehmen Sie an der J-PAL Evaluierungsstudie teil, um {{appName}} zu verbessern.",
```
Comment (klinzo) [inline translations/translations.json:5036]: Translations are from here if anyone wants to have a look, copied them manually: https://docs.google.com/document/d/1X7pfXlFT2M2E7zFVkMsul5rnzbANSusd/edit?usp=sharing&ouid=100822204933515470934&rtpof=true&sd=true

**Review by klinzo — COMMENTED** [review]
Please test this :)

File: native/src/components/Link.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React, { ReactElement } from 'react'
```
Comment (f1sh1918) [inline native/src/components/Link.tsx:1]: Just a general style issue. If i click on the link, background gets black. Looks a bit ugly ;)

File: native/src/routes/JpalTracking.tsx [hunk 4]
```diff
@@ -119,15 +162,19 @@ const JpalTracking = ({ route, theme }: PropsType) => {
...
         <Input
+          error={invalid && trackingCode?.length !== TRACKING_CODE_LENGTH}
           value={trackingCode ?? ''}
           onChangeText={updateTrackingCode}
           theme={theme}
           editable={!!trackingEnabled}
           testID='input'
         />
+
+        {error && <ErrorText>{t(error)}</ErrorText>}
```
Comment (SteffiStuffel) [inline native/src/routes/JpalTracking.tsx:173]: This won't translate the error as the namespace is missing. Should be changed to `{error && <ErrorText>{t(`error:${error}`)}</ErrorText>}`

File: native/src/routes/JpalTracking.tsx [hunk 5]
```diff
@@ -30,43 +32,94 @@ const DescriptionContainer = styled.TouchableOpacity`
...
+  }, [])
 
+  useEffect(
+    () =>
+      navigation.addListener('beforeRemove', e => {
+        if (!routeTrackingCode) {
+          // User probably just opened the settings out of interest, don't annoy em by opening an alert
+          return
+        }
+        if (trackingCode?.length === TRACKING_CODE_LENGTH && trackingEnabled) {
```
Comment (SteffiStuffel) [inline native/src/routes/JpalTracking.tsx:89]: So it you don't have a trackingCode in the url, it is fine to enter a TrackingCode with wrong lenght? Not sure if this is really wanted behaviour. Also would it make sense to give the user some feedback what the correct lenght would be?

File: native/src/routes/JpalTracking.tsx [hunk 6]
```diff
@@ -30,43 +32,94 @@ const DescriptionContainer = styled.TouchableOpacity`
   justify-content: space-between;
   padding: 15px 0 5px;
 `
-const Input = styled(TextInput)`
+const Input = styled(TextInput)<{ editable: boolean; error: boolean }>`
   padding: 15px;
   border-width: 1px;
   border-color: ${props => (props.editable ? props.theme.colors.themeColor : props.theme.colors.textDisabledColor)};
+  ${props => (props.error ? 'border-color: red;' : '')};
```
Comment (SteffiStuffel) [inline native/src/routes/JpalTracking.tsx:39]: Not working. Border is not red, although an error occured. See attached screenshot.

Comment (SteffiStuffel) [reply native/src/routes/JpalTracking.tsx:39]: Oke, only working when entering the jpal-settings-screen with a deep-link.

File: native/src/routes/JpalTracking.tsx (hunk 1 above)
Comment (SteffiStuffel) [reply native/src/routes/JpalTracking.tsx:96]: I get a wrong error message here:
Steps to reproduce:
1. Enter jpal-settings via deep link
2. keep switch in allowed-position but change tracking code to a code with wrong length
3. navigate back
4. Get the error that you should allow tracking, but it is allowed.

**Review by SteffiStuffel — COMMENTED** [review]
Tested on Android Emulator. The error message is hidden behind the keyboard, which is really not good.
When the border is red, a user can at least imagine that something went wrong, but when the border is not red, it just does not save the changes and the user gets no feedback.

Also not sure if it is good, that is saving of a change (switch or textfield) fails only the error is displayed, but the toggle/text-field stays in the new-input which is not saved. So it is not possible to try again and also as a user i wouldn't be sure if the changes are now saved or not, i  would expect it to be resetted to the old value.

IGAPP-628
IGAPP-628-hidden-error

File: native/src/components/Link.tsx (hunk 3 above)
Comment (klinzo) [reply native/src/components/Link.tsx:1]: can you take a screenshot/video?

File: native/src/routes/JpalTracking.tsx (hunk 4 above)
Comment (klinzo) [reply native/src/routes/JpalTracking.tsx]: It was included, see above. But not relevant anymore.

File: native/src/routes/JpalTracking.tsx (hunk 1 above)
Comment (klinzo) [reply native/src/routes/JpalTracking.tsx]: See new changes

File: native/src/routes/JpalTracking.tsx (hunk 4 above)
Comment (SteffiStuffel) [reply native/src/routes/JpalTracking.tsx]: Well, it was not working, I tested it.

Comment (klinzo) [reply native/src/routes/JpalTracking.tsx]: Was working for me :thinking:

**Comment by klinzo** [comment]
Still working on adding tests, maybe will do that in a separate PR

**Review by SteffiStuffel — APPROVED** [review]
Tested, JpalSettings are not visible when not opening via deep-link, but is visable and correctly opened and set, when opening via deep-link.

Things i did not like:
- When leaving the JPAL-Tracking-Site with tracking disabled, when clicking on "Allow tracking"-Button, the switch is now colored in red. I would just expect it to be toggled and be now yellow and in enabled-status.
- When disabling tracking the color of the switch is depending on the fact if the quit-modal was shown before or not, this is not really intuitiv. Either it should always or never be red in disabled position
- Now we have the solution, that the tracking code cannot be edited, what really makes sense, but it would be appropriate to display it (if we want to display it all) just as a text and not as a text-input, especially because the textfield does not really look disabled, but also will never be enabled and its a bit confusing.

**Comment by klinzo** [comment]
@SteffiStuffel could you have another look please? I figured we don't need to display the tracking code to the user at all. Tracking is now enabled on click on allow tracking.

**Review by f1sh1918 — COMMENTED** [review]
Tested on android, ios emulator and ios native.
Unfortunately i couldn't get the deep link work on ios native (Iphone se 2020)
https://integreat.app/jpal/123456789
The app opens but no redirect to jpal

**Review by f1sh1918 — APPROVED** [review]

File: native/src/components/Link.tsx (hunk 3 above)
Comment (f1sh1918) [reply native/src/components/Link.tsx:1]: Simulator Screen Recording - iPhone 12 Pro Max - 2021-11-09 at 17 42 00

**Comment by SteffiStuffel** [comment]
> Still working on adding tests, maybe will do that in a separate PR

Looks good.

File: native/src/routes/__tests__/JpalTracking.spec.tsx [hunk 7]
```diff
@@ -1,39 +1,80 @@
...
 import appSettings from '../../utils/AppSettings'
 import JpalTracking from '../JpalTracking'
 
 jest.mock('react-i18next')
+jest.mock('react-native', () => ({
+  ...jest.requireActual('react-native'),
+  Alert: {
+    alert: jest.fn()
+  }
+}))
```
Comment (Taggotty) [inline native/src/routes/__tests__/JpalTracking.spec.tsx:18]: Pls remove. Not needed.

File: native/src/components/Link.tsx (hunk 3 above)
Comment (Taggotty) [reply native/src/components/Link.tsx:1]: Could not reproduce this on android emulator o.o

**Comment by Taggotty** [comment]
teted on android emulator

File: native/src/components/Link.tsx (hunk 3 above)
Comment (klinzo) [reply native/src/components/Link.tsx:1]: Created https://issues.tuerantuer.org/browse/IGAPP-836

**Pull request merged**